### PR TITLE
Add script to test TiSpark

### PR DESCRIPTION
**Pull request by liufuyang**
To add another test to verify the tispark component can work.

So to have some early warning of errors mentioned here:
https://github.com/pingcap/tidb-docker-compose/issues/30

Current build job will fail due to issue mentioned at  https://github.com/pingcap/tidb-docker-compose/issues/30

@tennix Do you mind help re-run the test of this PR when the TiSpark docker image get updated?

### CODE REVIEW

**Comment by liufuyang** [comment]
image

Pretty sad that there is only a tag on tispark docker image, making me cannot roll back to an old tispark image version locally to test the tests.py script will actually work 😞

File: .travis.yml [hunk 1]
```diff
@@ -12,3 +12,5 @@ before_install:
 script:
   - docker ps -a --format="{{.Names}} {{.Image}} {{.Status}}" | grep -v 'Up' | grep -v 'Exited (0)' | awk '{print} END {if (NR>0) {exit 1;}}'
   - mysql -h 127.0.0.1 -P 4000 -u root -e "select tidb_version()\G" # test if tidb-server is working
+  - docker-compose exec tispark-master bash /opt/tispark/tests/loaddata.sh  # add some data for tests
```
Comment (tennix) [inline .travis.yml]: I think it's better to load data from host.
1. download and extract tispark-sample-data
2. load data from host
These two steps can be done in `before_install` section.

File: docker-compose.yml [hunk 2]
```diff
@@ -123,6 +123,7 @@ services:
       - /opt/spark/sbin/start-master.sh
     volumes:
       - ./config/spark-defaults.conf:/opt/spark/conf/spark-defaults.conf:ro
+      - ./tispark/tests:/opt/tispark/tests:ro
```
Comment (tennix) [inline docker-compose.yml]: We can start another container connecting to tidb-docker-compose network and run tests there without affecting tispark-master container.

File: .travis.yml (hunk 1 above)
Comment (tennix) [reply .travis.yml]: If we can load data from host, we are sure we can access TiDB, so no need to do the `select tidb_version()` step. And we don't need to add extra script to tispark-master container. But if you insist your option, I'm OK with it.

File: docker-compose-test.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  tispark-tests:
+    image: pingcap/tispark:latest
+    volumes:
+      - ./config/spark-defaults.conf:/opt/spark/conf/spark-defaults.conf:ro
+      - ./tispark-tests/tests:/opt/spark/tests:ro
+
+networks:
+  default:
+    external:
+      name: tidbdockercompose_default  # when travis'd docker version updates, use: tidb-docker-compose_default
```
Comment (tennix) [inline docker-compose-test.yml]: You can upgrade travis docker and docker-compose version as needed. https://docs.travis-ci.com/user/docker/#Installing-a-newer-Docker-version

**Comment by liufuyang** [comment]
@tennix finally it's working now. I had to update the docker-compose version as well to make it work with the newer way of network name. 
Also I think it is nice to have some db data in the tispark-master. It's pretty convenient for people to start containers from this repo then run an command to inject data. Unless we are going to remove it from another PR, then we can update the script to download the data from internet when doing the tests.

**Comment by liufuyang** [comment]
And let me know if you want me to squash everything into a single commit. 

Otherwise I think you can use the squash and merge option on the merge button. 

Looks like the tispark issue has been fixed, the build can all go through now.

**Comment by tennix** [comment]
Adding another docker-compose file for testing looks good.

**Pull request merged**